lapce-data/src/document.rs

```diff
@@ -2671,4 +2675,33 @@ impl Document {
         self.sticky_headers.borrow_mut().insert(line, lines.clone());
         lines
     }
+
+    pub fn change_syntax_selection(
+        &mut self,
+        direction: SelectionRangeDirection,
+    ) -> Option<Selection> {
+        if let Some(selections) = self.syntax_selection_range.as_mut() {
+            let range = match direction {
+                SelectionRangeDirection::Next => selections.next_range(),
+                SelectionRangeDirection::Previous => selections.previous_range(),
+            };
+
+            let region = range.map(|range| {
+                let start = self.buffer.offset_of_position(&range.start);
+                let end = self.buffer.offset_of_position(&range.end);
+                SelRegion::new(start, end, None)
+            });
+
+            let mut selection = Selection::new();
+            if let Some(region) = region {
```

Comment:
Would `region.map()` be a bit cleaner here? (Maybe not). Also, you create a `selection` one line up that you may not end up using.

Maybe this piece of code would look better as one single big `range.map(...)` or two maps.

To be clear, this is good enough, just a bit inconsistent in style :)